cli/src/main/java/com/devonfw/tools/ide/url/model/file/json/StatusJson.java

```diff
@@ -64,7 +66,37 @@ public void setUrls(Map<Integer, UrlStatus> urlStatuses) {
    */
   public UrlStatus getOrCreateUrlStatus(String url) {
 
+    UrlStatus urlStatus = this.urls.computeIfAbsent(computeKey(url), hash -> new UrlStatus());
+    urlStatus.markSillUsed();
+    return urlStatus;
+  }
+
+  static Integer computeKey(String url) {
     Integer key = Integer.valueOf(url.hashCode());
-    return this.urls.computeIfAbsent(key, hash -> new UrlStatus());
+    return key;
+  }
+
+  public void remove(String url) {
+
+    this.urls.remove(computeKey(url));
+  }
+
+  /**
+   * Performs a cleanup and removes all unused entries.
+   *
+   * @return {@code true} if something changed during cleanup, {@code false} otherwise.
+   */
+  public boolean cleanup() {
+
+    boolean changed = false;
+    Iterator<Entry<Integer, UrlStatus>> entryIterator = this.urls.entrySet().iterator();
+    while (entryIterator.hasNext()) {
+      Entry<Integer, UrlStatus> entry = entryIterator.next();
+      if (!entry.getValue().checkStillUsed()) {
```

Comment:
unused entries were never removed before this change ending up as dead JSON causing confusion.
Really hard to understand what was going on since the JSON keys are only cryptic hashes of the URLs.
So if you have 5 URLs and 11 entries in JSON you get puzzled but have no idea which entry was right and which is wrong. Some entries were (are currently) entirely empty so they neither contain `success` nor `error`.